### PR TITLE
Add slim agent

### PR DESCRIPTION
**Pull request by mstemm**
Changes to support upcoming variants of the sysdig agent that split building kernel module and running the agent.

### CODE REVIEW

**Comment by mstemm** [comment]
@bencer could you take a look?

File: agent_deploy/kubernetes/sysdig-agent-slim-daemonset-v1.yaml [hunk 1]
```diff
@@ -0,0 +1,125 @@
...
+        imagePullPolicy: Always
+        securityContext:
+          privileged: true
+        resources:
+          # Resources needed are subjective on the actual workload
+          # please refer to Sysdig Support for more info about it
+          requests:
+            cpu: 100m
+            memory: 512Mi
+          limits:
```
Comment (mbreitung) [inline agent_deploy/kubernetes/sysdig-agent-slim-daemonset-v1.yaml]: no limit on cpu?

File: agent_deploy/kubernetes/sysdig-agent-slim-daemonset-v2.yaml [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+        imagePullPolicy: Always
+        securityContext:
+          privileged: true
+        resources:
+          # Resources needed are subjective on the actual workload
+          # please refer to Sysdig Support for more info about it
+          requests:
+            cpu: 100m
+            memory: 512Mi
+          limits:
```
Comment (mbreitung) [inline agent_deploy/kubernetes/sysdig-agent-slim-daemonset-v2.yaml]: no limit on cpu?

File: agent_deploy/kubernetes/sysdig-agent-slim-daemonset-v2.yaml [hunk 3]
```diff
@@ -0,0 +1,109 @@
...
+            memory: 1024Mi
```
Comment (mbreitung) [inline agent_deploy/kubernetes/sysdig-agent-slim-daemonset-v2.yaml]: does the init container need so many resources?

File: agent_deploy/kubernetes/sysdig-agent-slim-daemonset-v1.yaml (hunk 1 above)
Comment (mstemm) [reply agent_deploy/kubernetes/sysdig-agent-slim-daemonset-v1.yaml]: none of the other variants have a limit on cpu, as far as I know.

File: agent_deploy/kubernetes/sysdig-agent-slim-daemonset-v2.yaml (hunk 3 above)
Comment (mstemm) [reply agent_deploy/kubernetes/sysdig-agent-slim-daemonset-v2.yaml]: I didn't carefully figure out what resources the builder image uses, but it's short lived so I'm not sure it's that important. I'd rather use these higher limits than pick a limit that is too low.

**Comment by bencer** [comment]
this looks good to me, there is the separated issue of using requests/limits with more reasonable values but that shouldn't affect this merge imho

**Comment by mstemm** [comment]
I tuned the cpu and memory limits a bit for the agent-kmodule container.

Is this ready to merge? We're going to start publishing the alternate agent images soon.

**Pull request merged**